tests/functional/src/main/java/org/mule/functional/junit4/runners/ArtifactClassLoaderRunnerConfig.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.functional.junit4.runners;
+
+import org.mule.runtime.extension.api.annotation.Extension;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Specifies a configuration needed by {@link ArtifactClassloaderTestRunner} in order to
+ * run the test.
+ *
+ * @since 4.0
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+@Inherited
+public @interface ArtifactClassLoaderRunnerConfig
+{
+
+    /**
+     * @return a comma separated list of packages to be added as PARENT_ONLY for the
+     * container class loader, default (and required) packages is empty.
+     * <p/>
+     * A default list of extra boot packages is always added to the container class loader, it is defined by the excluded.properties file.
+     * <p/>
+     * In case if a particular test needs to add extra boot packages to append to the ones already defined in the file, it will
+     * have to define it here by using this annotation method.
+     */
+    String extraBootPackages() default "";
+
+    /**
+     * @return {@link String} to define the base package to be used when discovering for extensions in order to create for each
+     * extension a plugin {@link ClassLoader}. If no extension are discovered plugin class loaders will not be created.
+     * The discovery process will look for classes in classpath annotated with {@link Extension}.
+     */
+    String extensionBasePackage() default "org.mule.extension";
```

Comment:
The thing about this is, if I don't set a base package it will take a LOT to discover the classes annotated in the whole classpath. I know that it will be an issue but it has to be set in order to avoid wasting a lot of time while doing this discovery. It is a trade-off between declaring the extension vs auto-discovering process.